inference-engine/samples/speech_sample/main.cpp

```diff
@@ -601,43 +601,47 @@ int main(int argc, char *argv[]) {
-                std::vector<uint8_t> ptrFeatures;
-                uint32_t numArrays(0), numBytes(0), numFrames(0), numFrameElements(0), numBytesPerElement(0);
-                GetKaldiArkInfo(inputArkName, 0, &numArrays, &numBytes);
-                ptrFeatures.resize(numBytes);
-                LoadKaldiArkArray(inputArkName,
-                                  0,
-                                  name,
-                                  ptrFeatures,
-                                  &numFrames,
-                                  &numFrameElements,
-                                  &numBytesPerElement);
-                auto floatScaleFactor =
+                for (size_t i = 0; i < scaleFactorInput.size(); ++i) {
+                    slog::info << "For input " << i << " using scale factor of " << scaleFactorInput[i] << slog::endl;
+                    std::string scaleFactorConfigKey = GNA_CONFIG_KEY(SCALE_FACTOR) + std::string("_") + std::to_string(i);
+                    gnaPluginConfig[scaleFactorConfigKey] = scaleFactorInput[i];
+                }
+            } else {
+                // "static" quantization with calculated scale factor
+                for (size_t i = 0; i < numInputArkFiles; i++) {
+                    auto inputArkName = inputArkFiles[i].c_str();
+                    std::string name;
+                    std::vector<uint8_t> ptrFeatures;
+                    uint32_t numArrays(0), numBytes(0), numFrames(0), numFrameElements(0), numBytesPerElement(0);
+                    GetKaldiArkInfo(inputArkName, 0, &numArrays, &numBytes);
+                    ptrFeatures.resize(numBytes);
+                    LoadKaldiArkArray(inputArkName,
+                        0,
+                        name,
+                        ptrFeatures,
+                        &numFrames,
+                        &numFrameElements,
+                        &numBytesPerElement);
+                    auto floatScaleFactor =
                         ScaleFactorForQuantization(ptrFeatures.data(), MAX_VAL_2B_FEAT, numFrames * numFrameElements);
-                slog::info << "Using scale factor of " << floatScaleFactor << " calculated from first utterance."
-                           << slog::endl;
-                std::string scaleFactorConfigKey = GNA_CONFIG_KEY(SCALE_FACTOR) + std::string("_") + std::to_string(i);
-                gnaPluginConfig[scaleFactorConfigKey] = std::to_string(floatScaleFactor);
+                    slog::info << "Using scale factor of " << floatScaleFactor << " calculated from first utterance."
+                        << slog::endl;
+                    std::string scaleFactorConfigKey = GNA_CONFIG_KEY(SCALE_FACTOR) + std::string("_") + std::to_string(i);
+                    gnaPluginConfig[scaleFactorConfigKey] = std::to_string(floatScaleFactor);
+                }
             }
+        } else {
+            slog::info << "Using scale factor from provided exported gna model: " << FLAGS_rg << slog::endl;
```

Comment:
I think we should warn the user f the scale factors were set through the command line when exporting a model, not only log it.
If everything's fine, I am not sure it makes sense to report this at all.

Reply:
done